File: packages/quick-image/ios/QuickImageModule.swift
import ExpoModulesCore
import SDWebImage

public class QuickImageModule: Module {
  public func definition() -> ModuleDefinition {
    Name("QuickImage")

    ViewManager {
      View {
        QuickImageView()
      }

      Prop("uri") { (view: QuickImageView, prop: URL) in
        view.sd_imageIndicator = SDWebImageActivityIndicator.gray;
        view.sd_imageTransition = SDWebImageTransition.fade;
        view.sd_setImage(with: prop)
        view.contentMode = UIView.ContentMode.scaleAspectFill
      }
    }
  }
}

File: packages/quick-image/android/src/main/java/expo/modules.quickimage/QuickImageModule.kt
package expo.modules.quickimage

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class QuickImageModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("QuickImage")

    AsyncFunction("helloAsync") { options: Map<String, String> ->
      println("Hello 👋")
    }

    ViewManager {
      View { context -> 
        QuickImageView(context) 
      }

      Prop("name") { view: QuickImageView, prop: String ->
        println(prop)
      }
    }
  }
}
